pkg/clients/clients.go

```diff
@@ -148,12 +148,15 @@ func (dcs *DynamicClientSet) ResourceClientForObject(obj *unstructured.Unstructu
 	return dcs.ResourceClient(obj.GroupVersionKind(), obj.GetNamespace())
 }
 
-func (dcs *DynamicClientSet) gvkForKind(kind Kind) (gvk schema.GroupVersionKind, err error) {
-	resources, err := dcs.getServerResources()
-	if err != nil && len(resources) == 0 {
-		// Only return an error here if no resources were returned. Otherwise, process the partial list
-		// and return an error if no match is found.
-		return
+func (dcs *DynamicClientSet) gvkForKind(kind Kind) (*schema.GroupVersionKind, error) {
+	resources, err := dcs.DiscoveryClientCached.ServerPreferredResources()
+	if err != nil {
+		if discovery.IsGroupDiscoveryFailedError(err) {
+			// Ignore Group Discovery Failed errors. This type of error may cause
+			// later resource introspection to fail, but the error will be handled at that point.
```

Comment:
For my own education: in this case, `resources` is empty, right? Also, `the error will be handled at that point` is referring to the return at the end of this method right? We might just reword it to say that instead -- not sure what it means as written.

Reply:
I'll clarify the comment. `ServerPreferredResources` actually returns a partial list in case discovery failed for one or more groups. This ignores those errors and attempts to process the partial list. If the discovery failed for a relevant group, introspection will fail for the caller.